eppoclient/rules.go

```diff
@@ -227,15 +195,61 @@ func evaluateNumericCondition(subjectValue float64, conditionValue float64, cond
 // Returns a float64 and nil error on success, or 0 and an error on failure.
 func toFloat64(val interface{}) (float64, error) {
 	switch v := val.(type) {
-	case float64:
-		return v, nil
+	case float32, float64:
+		return promoteFloat(v), nil 
+	case int, int8, int16, int32, int64:
+		return float64(promoteInt(v)), nil
+	case uint, uint8, uint16, uint32, uint64:
+		return float64(promoteUint(v)), nil
 	case string:
 		floatVal, err := strconv.ParseFloat(v, 64)
 		if err != nil {
 			return 0, fmt.Errorf("cannot convert string '%s' to float64: %w", v, err)
 		}
 		return floatVal, nil
 	default:
-		return 0, errors.New("value is neither a float64 nor a convertible string")
+		return 0, errors.New("value is neither a number nor a convertible string")
+	}
+}
+
+func promoteInt(i interface{}) int64 {
+	switch i := i.(type) {
+	case int:
+		return int64(i)
+	case int8:
+		return int64(i)
+	case int16:
+		return int64(i)
+	case int32:
+		return int64(i)
+	case int64:
+		return i
+	}
+	panic(fmt.Errorf("unexpected type passed to promoteInt: %T", i))
+}
+
+func promoteUint(i interface{}) uint64 {
+	switch i := i.(type) {
+	case uint:
+		return uint64(i)
+	case uint8:
+		return uint64(i)
+	case uint16:
+		return uint64(i)
+	case uint32:
+		return uint64(i)
+	case uint64:
+		return i
+	}
+	panic(fmt.Errorf("unexpected type passed to promoteUint: %T", i))
+}
+
+func promoteFloat(f interface{}) float64 {
```

Comment:
Newly added simple function to keep our `case` statement in `toFloat64()` cleaner